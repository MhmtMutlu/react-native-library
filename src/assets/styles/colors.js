const colors = {
  grey: '#6c757d',
  darkGrey: '#495057',
  darkGrey2x: '#343a40',
  lightGrey: '#adb5bd',
  lightGrey2x: '#dee2e6',
  lightGrey3x: '#e0e0e0',
  white: '#f8f9fa',
};

export default colors;
