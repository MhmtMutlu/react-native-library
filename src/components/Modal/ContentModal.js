import React from 'react';
import {View, TextInput} from 'react-native';
import Button from '../Button';
import Modal from 'react-native-modal';
import styles from './ContentModal.styles';
import {Formik} from 'formik';

const initialValues = {
  bookName: '',
  writer: '',
  bookSubject: '',
  bookPages: 0,
  quotes: '',
  startDate: '',
  endDate: '',
  imageUrl: '',
};

const ContentModal = ({isVisible, onSend, onClose}) => {
  return (
    <Modal
      style={styles.modal}
      isVisible={isVisible}
      swipeDirection="down"
      onSwipeComplete={onClose}
      onBackdropPress={onClose}
      onBackButtonPress={onClose}>
      <View style={styles.container}>
        <Formik initialValues={initialValues} onSubmit={onSend}>
          {({handleSubmit, handleChange, values}) => (
            <View style={styles.body_container}>
              <View style={styles.input}>
                <TextInput
                  value={values.bookName}
                  placeholder="Kitap adı ..."
                  onChangeText={handleChange('bookName')}
                />
              </View>
              <View style={styles.input}>
                <TextInput
                  value={values.writer}
                  placeholder="Yazarı ..."
                  onChangeText={handleChange('writer')}
                />
              </View>
              <View style={styles.input}>
                <TextInput
                  value={values.bookSubject}
                  placeholder="Kitabın konusu ..."
                  onChangeText={handleChange('bookSubject')}
                />
              </View>
              <View style={styles.input}>
                <TextInput
                  value={values.quotes}
                  placeholder="Alıntılar ..."
                  onChangeText={handleChange('quotes')}
                />
              </View>
              <View style={styles.input}>
                <TextInput
                  value={values.startDate}
                  placeholder="Başlangıç tarihi ..."
                  onChangeText={handleChange('startDate')}
                />
              </View>
              <View style={styles.input}>
                <TextInput
                  value={values.endDate}
                  placeholder="Başlangıç tarihi ..."
                  onChangeText={handleChange('endDate')}
                />
              </View>
              <View style={styles.input}>
                <TextInput
                  value={values.imageUrl}
                  placeholder="Kitabın resmi ..."
                  onChangeText={handleChange('imageUrl')}
                />
              </View>
              <View style={styles.button_wrapper}>
                <Button text="Giriş Yap" onPress={handleSubmit} />
                <Button
                  text="Kaydet"
                  theme="secondary"
                  onPress={handleSubmit}
                />
              </View>
            </View>
          )}
        </Formik>
      </View>
    </Modal>
  );
};

export default ContentModal;
