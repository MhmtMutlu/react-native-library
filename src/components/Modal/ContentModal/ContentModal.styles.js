import {StyleSheet, Dimensions} from 'react-native';
import colors from '../../../assets/styles/colors';

const deviceSize = Dimensions.get('window');

export default StyleSheet.create({
  container: {
    backgroundColor: colors.white,
    paddingTop: 15,
    padding: 5,
    borderRadius: 15,
    height: deviceSize.height / 1.4,
  },
  modal: {
    left: deviceSize.width / 100,
    right: deviceSize.width / 100,
    position: 'absolute',
    top: 15,
    marginVertical: 20,
  },
  input: {
    borderRadius: 10,
    marginBottom: 10,
    backgroundColor: colors.lightGrey3x,
  },
  date_wrapper: {
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
  },
  date: {
    flex: 0.5,
  },
  button_wrapper: {
    marginTop: 10,
  },
});
