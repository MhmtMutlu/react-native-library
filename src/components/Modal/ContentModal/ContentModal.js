import React, {useState} from 'react';
import {View} from 'react-native';
import {Formik} from 'formik';
import Modal from 'react-native-modal';
import DatePicker from 'react-native-date-picker';

import Input from '../../Input';
import Button from '../../Button';

import styles from './ContentModal.styles';

const ContentModal = ({isVisible, onUpdate, onClose, onDelete, itemValues}) => {
  const initialValues = {...itemValues[0]};
  const [firstDateVisibility, setFirstDateVisibility] = useState(false);
  const [secondDateVisibility, setSecondDateVisibility] = useState(false);

  return (
    <Modal
      testID="test-content-modal"
      coverScreen={true}
      style={styles.modal}
      isVisible={isVisible}
      swipeDirection="down"
      onSwipeComplete={onClose}
      onBackdropPress={onClose}
      onBackButtonPress={onClose}>
      <View testID="test-content-modal-container" style={styles.container}>
        <Formik
          initialValues={initialValues}
          onSubmit={value => onUpdate(value, initialValues.id)}>
          {({handleSubmit, handleChange, setFieldValue, values}) => (
            <View style={styles.body_container}>
              <Input
                value={values.bookName}
                placeholder="Kitap adı ..."
                onType={handleChange('bookName')}
              />
              <Input
                value={values.writer}
                placeholder="Yazarı ..."
                onType={handleChange('writer')}
              />
              <Input
                isMultiline={true}
                value={values.bookSubject}
                placeholder="Kitabın konusu ..."
                onType={handleChange('bookSubject')}
              />
              <Input
                isMultiline={true}
                value={values.quotes}
                placeholder="Alıntılar ..."
                onType={handleChange('quotes')}
              />
              <View style={styles.date_wrapper}>
                <View style={styles.date}>
                  <Button
                    text="Başlangıç Tarihi"
                    theme="secondary"
                    onPress={() => setFirstDateVisibility(true)}
                  />
                  <DatePicker
                    modal
                    androidVariant="iosClone"
                    open={firstDateVisibility}
                    mode="date"
                    date={new Date(values.startDate)}
                    onConfirm={date => {
                      setFirstDateVisibility(false);
                      setFieldValue('startDate', date);
                    }}
                    onCancel={() => {
                      setFirstDateVisibility(false);
                    }}
                  />
                </View>
                <View style={styles.date}>
                  <Button
                    text="Bitiş Tarihi"
                    theme="secondary"
                    onPress={() => setSecondDateVisibility(true)}
                  />
                  <DatePicker
                    modal
                    androidVariant="iosClone"
                    open={secondDateVisibility}
                    mode="date"
                    date={new Date(values.endDate)}
                    onConfirm={date => {
                      setSecondDateVisibility(false);
                      setFieldValue('endDate', date);
                    }}
                    onCancel={() => {
                      setSecondDateVisibility(false);
                    }}
                  />
                </View>
              </View>
              <View style={styles.button_wrapper}>
                <Button text="Güncelle" onPress={handleSubmit} />
                <Button
                  text="Sil"
                  theme="secondary"
                  onPress={() => onDelete(initialValues.id)}
                />
              </View>
            </View>
          )}
        </Formik>
      </View>
    </Modal>
  );
};

export default ContentModal;
