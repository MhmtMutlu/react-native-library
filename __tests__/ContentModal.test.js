import React from 'react';
import {render} from '@testing-library/react-native';
import ContentModal from '../src/components/Modal/ContentModal';
import styles from '../src/components/Modal/ContentModal/ContentModal.styles';

const dummyData = [
  {
    bookName: 'test book',
    writer: 'test writer',
    bookSubject: 'test subject',
    endDate: '2020-05-25T21:27:00.000Z',
    startDate: '2020-05-25T21:27:00.000Z',
    quotes: 'test quotes',
  },
];

test('should match with snapshot', () => {
  const comp = render(<ContentModal isVisible={true} itemValues={dummyData} />);
  expect(comp).toMatchSnapshot();
});

test('check visible prop', () => {
  const comp = render(<ContentModal isVisible={true} itemValues={dummyData} />);
  const contentModal = comp.getByTestId('test-content-modal').props;
  expect(contentModal).toMatchObject({
    visible: true,
  });
});

test('should render ContentModal css', () => {
  const comp = render(<ContentModal isVisible={true} itemValues={dummyData} />);
  const contentModal = comp.getByTestId('test-content-modal-container').props
    .style;
  expect(contentModal).toMatchSnapshot(styles.container);
});
