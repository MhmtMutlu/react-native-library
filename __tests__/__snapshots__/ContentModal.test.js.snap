// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match with snapshot 1`] = `
<Modal
  animationType="none"
  deviceHeight={null}
  deviceWidth={null}
  hardwareAccelerated={false}
  hideModalContentWhileAnimating={false}
  onBackdropPress={[Function]}
  onModalHide={[Function]}
  onModalWillHide={[Function]}
  onModalWillShow={[Function]}
  onRequestClose={[Function]}
  panResponderThreshold={4}
  scrollHorizontal={false}
  scrollOffset={0}
  scrollOffsetMax={0}
  scrollTo={null}
  statusBarTranslucent={false}
  supportedOrientations={
    Array [
      "portrait",
      "landscape",
    ]
  }
  swipeDirection="down"
  swipeThreshold={100}
  testID="test-content-modal"
  transparent={true}
  visible={true}
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    forwardedRef={[Function]}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "black",
        "bottom": 0,
        "height": 1334,
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 750,
      }
    }
  />
  <View
    collapsable={false}
    deviceHeight={null}
    deviceWidth={null}
    forwardedRef={[Function]}
    hideModalContentWhileAnimating={false}
    onBackdropPress={[Function]}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    panResponderThreshold={4}
    pointerEvents="box-none"
    scrollHorizontal={false}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    statusBarTranslucent={false}
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
        "left": 7.5,
        "margin": 37.5,
        "marginVertical": 20,
        "position": "absolute",
        "right": 7.5,
        "top": 15,
        "transform": Array [
          Object {
            "translateY": 1334,
          },
        ],
      }
    }
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeDirection="down"
    swipeThreshold={100}
  >
    <View
      style={
        Object {
          "backgroundColor": "#f8f9fa",
          "borderRadius": 15,
          "height": 952.8571428571429,
          "padding": 5,
          "paddingTop": 15,
        }
      }
      testID="test-content-modal-container"
    >
      <View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#e0e0e0",
              "borderRadius": 10,
              "color": "#343a40",
              "flexDirection": "row",
              "justifyContent": "center",
              "margin": 5,
              "paddingHorizontal": 10,
              "paddingVertical": 1,
            }
          }
          testID="input-view"
        >
          <TextInput
            multiline={false}
            numberOfLines={null}
            onChangeText={[Function]}
            placeholder="Kitap adı ..."
            placeholderTextColor="#343a40"
            style={
              Object {
                "flex": 1,
              }
            }
            testID="text-input"
            textAlignVertical={null}
            value="test book"
          />
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              Array [
                Object {
                  "color": "#343a40",
                  "fontSize": 25,
                },
                undefined,
                Object {
                  "fontFamily": "Material Design Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#e0e0e0",
              "borderRadius": 10,
              "color": "#343a40",
              "flexDirection": "row",
              "justifyContent": "center",
              "margin": 5,
              "paddingHorizontal": 10,
              "paddingVertical": 1,
            }
          }
          testID="input-view"
        >
          <TextInput
            multiline={false}
            numberOfLines={null}
            onChangeText={[Function]}
            placeholder="Yazarı ..."
            placeholderTextColor="#343a40"
            style={
              Object {
                "flex": 1,
              }
            }
            testID="text-input"
            textAlignVertical={null}
            value="test writer"
          />
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              Array [
                Object {
                  "color": "#343a40",
                  "fontSize": 25,
                },
                undefined,
                Object {
                  "fontFamily": "Material Design Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#e0e0e0",
              "borderRadius": 10,
              "color": "#343a40",
              "flexDirection": "row",
              "justifyContent": "center",
              "margin": 5,
              "paddingHorizontal": 10,
              "paddingVertical": 1,
            }
          }
          testID="input-view"
        >
          <TextInput
            multiline={true}
            numberOfLines={6}
            onChangeText={[Function]}
            placeholder="Kitabın konusu ..."
            placeholderTextColor="#343a40"
            style={
              Object {
                "flex": 1,
              }
            }
            testID="text-input"
            textAlignVertical="top"
            value="test subject"
          />
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              Array [
                Object {
                  "color": "#343a40",
                  "fontSize": 25,
                },
                undefined,
                Object {
                  "fontFamily": "Material Design Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#e0e0e0",
              "borderRadius": 10,
              "color": "#343a40",
              "flexDirection": "row",
              "justifyContent": "center",
              "margin": 5,
              "paddingHorizontal": 10,
              "paddingVertical": 1,
            }
          }
          testID="input-view"
        >
          <TextInput
            multiline={true}
            numberOfLines={6}
            onChangeText={[Function]}
            placeholder="Alıntılar ..."
            placeholderTextColor="#343a40"
            style={
              Object {
                "flex": 1,
              }
            }
            testID="text-input"
            textAlignVertical="top"
            value="test quotes"
          />
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              Array [
                Object {
                  "color": "#343a40",
                  "fontSize": 25,
                },
                undefined,
                Object {
                  "fontFamily": "Material Design Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <View
            style={
              Object {
                "flex": 0.5,
              }
            }
          >
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#e0e0e0",
                  "borderRadius": 10,
                  "margin": 5,
                  "opacity": 1,
                  "padding": 10,
                }
              }
              testID="button-touchable"
            >
              <Text
                style={
                  Object {
                    "color": "#343a40",
                    "fontSize": 17,
                    "fontWeight": "bold",
                  }
                }
                testID="button-text"
              >
                Başlangıç Tarihi
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "flex": 0.5,
              }
            }
          >
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#e0e0e0",
                  "borderRadius": 10,
                  "margin": 5,
                  "opacity": 1,
                  "padding": 10,
                }
              }
              testID="button-touchable"
            >
              <Text
                style={
                  Object {
                    "color": "#343a40",
                    "fontSize": 17,
                    "fontWeight": "bold",
                  }
                }
                testID="button-text"
              >
                Bitiş Tarihi
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "marginTop": 10,
            }
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#343a40",
                "borderRadius": 10,
                "margin": 5,
                "opacity": 1,
                "padding": 10,
              }
            }
            testID="button-touchable"
          >
            <Text
              style={
                Object {
                  "color": "#f8f9fa",
                  "fontSize": 17,
                  "fontWeight": "bold",
                }
              }
              testID="button-text"
            >
              Güncelle
            </Text>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#e0e0e0",
                "borderRadius": 10,
                "margin": 5,
                "opacity": 1,
                "padding": 10,
              }
            }
            testID="button-touchable"
          >
            <Text
              style={
                Object {
                  "color": "#343a40",
                  "fontSize": 17,
                  "fontWeight": "bold",
                }
              }
              testID="button-text"
            >
              Sil
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;

exports[`should render ContentModal css 1`] = `
Object {
  "backgroundColor": "#f8f9fa",
  "borderRadius": 15,
  "height": 952.8571428571429,
  "padding": 5,
  "paddingTop": 15,
}
`;
